pageserver/src/failpoint_support.rs

```diff
@@ -34,6 +34,44 @@ pub(crate) async fn failpoint_sleep_helper(name: &'static str, duration_str: Str
     tracing::info!("failpoint {:?}: sleep done", name);
 }
 
+/// Declare a failpoint that can use the `pause` failpoint action.
+/// We don't want to block the executor thread, hence, spawn_blocking + await.
+#[macro_export]
+macro_rules! __pausable_failpoint {
+    ($name:literal) => {
+        if cfg!(feature = "testing") {
+            let barrier = std::sync::Arc::new(tokio::sync::Barrier::new(2));
+            let mut jh = tokio::task::spawn_blocking({
+                let current = tracing::Span::current();
+                let barrier = barrier.clone();
+                move || {
+                    let _entered = current.entered();
+                    // pausing will park the thread within `fail`, but we want to synchronize with
+                    // failpoints from test code which we currently do by waiting for a log line.
+                    //
+                    // instead of producing the log line conditionally, produce it only after
+                    // 100ms of joining at the barrier.
+                    tokio::runtime::Handle::current().block_on(barrier.wait());
```

Comment:
Haven't we banned this function?

Reply:
I don't see clippies. Do you have an alternative proposal for implementing this?

Reply:
It's currently disabled because there is uses of it left due to the asyncification not having been completed (for example 41ee75bc7149688659999c429e8da6847e3e0f36). See https://github.com/neondatabase/neon/pull/5101.

Reply:
This call is within spawn_blocking and there is no other way to implement this.